packages/babel-plugin-transform-exponentiation-operator/src/index.ts

```diff
@@ -1,26 +1,87 @@
 import { declare } from "@babel/helper-plugin-utils";
-import build from "@babel/helper-builder-binary-assignment-operator-visitor";
-import { types as t } from "@babel/core";
+import type { types as t, Scope } from "@babel/core";
 
 export default declare(api => {
   api.assertVersion(REQUIRED_VERSION(7));
 
+  const { types: t } = api;
+
+  function build(left: t.Expression, right: t.Expression) {
+    return t.callExpression(
+      t.memberExpression(t.identifier("Math"), t.identifier("pow")),
+      [left, right],
+    );
+  }
+
+  function maybeMemoize(node: t.Expression, scope: Scope) {
+    if (scope.isPure(node)) return { assign: node, ref: t.cloneNode(node) };
+
+    const id = scope.generateUidIdentifierBasedOnNode(node);
+    scope.push({ id });
+    return {
+      assign: t.assignmentExpression("=", t.cloneNode(id), node),
+      ref: t.cloneNode(id),
+    };
+  }
+
   return {
     name: "transform-exponentiation-operator",
 
-    visitor: build({
-      operator: "**",
-
-      build(left, right) {
-        return t.callExpression(
-          t.memberExpression(t.identifier("Math"), t.identifier("pow")),
-          [
-            // left can be PrivateName only if operator is `"in"`
-            left as t.Expression,
-            right,
-          ],
-        );
+    visitor: {
+      AssignmentExpression(path) {
+        const { node, scope } = path;
+        if (node.operator !== "**=") return;
+
+        if (t.isMemberExpression(node.left)) {
+          let member1: t.Expression;
+          let member2: t.Expression;
+
+          const object = maybeMemoize(node.left.object, scope);
+          const { property, computed } = node.left;
+
+          if (computed) {
+            const prop = maybeMemoize(property as t.Expression, scope);
+            member1 = t.memberExpression(object.assign, prop.assign, true);
+            member2 = t.memberExpression(object.ref, prop.ref, true);
+          } else {
+            member1 = t.memberExpression(object.assign, property, false);
+            member2 = t.memberExpression(
+              object.ref,
+              t.cloneNode(property),
+              false,
+            );
+          }
+
+          path.replaceWith(
+            t.assignmentExpression("=", member1, build(member2, node.right)),
+          );
+        } else {
+          path.replaceWith(
+            t.assignmentExpression(
+              "=",
+              node.left,
+              build(
+                // todo: it could be a t.AsExpression
```

Comment:
This todo was already there in the original implementation, and nobody ever discovered it :grimacing: